kythe/extractors/openjdk11/extract.go

```diff
@@ -0,0 +1,196 @@
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package main
+
+import (
+	"bufio"
+	"errors"
+	"flag"
+	"io"
+	"log"
+	"os"
+	"os/exec"
+	"path/filepath"
+	"regexp"
+	"strings"
+
+	"kythe.io/kythe/go/util/flagutil"
+
+	"github.com/bazelbuild/rules_go/go/tools/bazel"
+)
+
+const (
+	javaCommandVar      = "KYTHE_JAVA_COMMAND"
+	kytheRootVar        = "KYTHE_ROOT_DIRECTORY"
+	kytheVnameVar       = "KYTHE_VNAMES"
+	javaMakeVar         = "JAVA_CMD"
+	runfilesWrapperPath = "io_kythe/kythe/extractors/openjdk11/java-wrapper"
+	runfilesVnamesPath  = "io_kythe/kythe/extractors/openjdk11/vnames.json"
+)
+
+var (
+	wrapperPath  string
+	vnameRules   string
+	errorPattern = regexp.MustCompile("ERROR: extractor failure for module ([^:]*):")
+)
+
+func setupRunfiles() error {
+	if len(os.Getenv("RUNFILES_DIR")) > 0 || len(os.Getenv("RUNFILES_MANIFEST_FILE")) > 0 {
+		return nil
+	}
+	for _, base := range []string{os.Args[0] + ".runfiles", "."} {
+		if root, err := filepath.Abs(base); err == nil {
+			if _, err := os.Stat(root); err == nil {
+				os.Setenv("RUNFILES_DIR", root)
+				if _, err := os.Stat(filepath.Join(root, "MANIFEST")); err == nil {
+					os.Setenv("RUNFILES_MANIFEST_FILE", filepath.Join(root, "MANIFEST"))
+				}
+				return nil
+			}
+		}
+	}
+	return errors.New("unable to setup runfiles")
+}
+
+func defaultWrapperPath() string {
+	val, _ := bazel.Runfile(runfilesWrapperPath)
+	return val
+}
+
+func defaultVnamesPath() string {
+	val, _ := bazel.Runfile(runfilesVnamesPath)
+	return val
+}
+
+func makeDir() string {
+	if dir := flag.Arg(0); len(dir) > 0 {
+		return dir
+	}
+	return "."
+}
+
+func findJavaCommand() (string, error) {
+	cmd := exec.Command("make", "-n", "-p")
+	cmd.Dir = makeDir()
+	cmd.Stderr = os.Stderr
+	stdout, err := cmd.StdoutPipe()
+	if err != nil {
+		return "", err
+	}
+	if err := cmd.Start(); err != nil {
+		return "", err
+	}
+	prefix := javaMakeVar + " := "
+	scanner := bufio.NewScanner(stdout)
+	for scanner.Scan() {
+		if strings.HasPrefix(scanner.Text(), prefix) {
+			stdout.Close()
+			cmd.Wait()
+			return strings.TrimPrefix(scanner.Text(), prefix), nil
+		}
+	}
+	return "", cmd.Wait()
+}
+
+func mustFindJavaCommand() string {
+	java, err := findJavaCommand()
+	if err != nil {
+		log.Fatalf("unable to determine KYTHE_JAVA_COMMAND: %v", err)
+	}
+	return java
+}
+
+func setenvDefaultFunc(env []string, key string, value func() string) []string {
```

Comment:
`Getenv` is ironically not an idiomatic name.  We should not follow its bad example; instead use `setEnv...`.